src/devices/Hx711/HX711.cs

```diff
@@ -0,0 +1,274 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Device;
+using System.Device.Gpio;
+using System.Linq;
+using UnitsNet;
+
+namespace Iot.Device.Hx711
+{
+    /// <summary>
+    /// Hx711 - Weight scale Module
+    /// </summary>
+    /// <remarks>DataSheet: https://html.alldatasheet.com/html-pdf/1132222/AVIA/Hx711/109/1/Hx711.html</remarks>
+    public sealed class Hx711 : IDisposable
+    {
+        private readonly GpioController _gpioController;
+        private readonly bool _shouldDispose;
+
+        private readonly int _pinDout;
+        private readonly int _pinPdSck;
+
+        private readonly Hx711Options _options;
+        private readonly object _readLock;
+        private readonly Hx711Reader _reader;
+
+        private readonly List<double> _referenceUnitList = new();
+
+        private bool _isInitialize;
+        private bool _isCalibrated;
+        private int _tareValue;
+
+        /// <summary>
+        /// Offset value from 0 at startup
+        /// </summary>
+        private int _offsetFormZero;
+
+        /// <summary>
+        /// Conversion ratio between Hx711 units and grams
+        /// </summary>
+        public double ReferanceUnit { get; private set; }
+
+        /// <summary>
+        /// Weight set as tare
+        /// </summary>
+        public Mass TareValue
+        {
+            get
+            {
+                return ReferanceUnit == 0 ? Mass.FromGrams(_tareValue) : Mass.FromGrams(Math.Round(_tareValue / ReferanceUnit));

```

Comment:
why Math.Round?

Reply:
Because `ReferenceUnit` value can be a number with a comma and applied as a divisor to the tare or weight value it can return a value with many digits.
In some cases it is possible to have started from 200g, in the calibration process, if you re-weigh the same object it could result in 199.99999999998.
Rounding serves precisely to always have a number of grams (see the same thing in `GetWeight`).
I figured in most cases you will have load cells that weigh grams (or kilograms) and you don't want to weigh nanograms.
If this were the case, the management of the staircase to be used must be added, not thought of at the moment.

Reply:
IMO you should not do rounding - users can do round themselves. If you add enough samples your accuracy should be less than a gram and doing rounding kills that

Reply:
ok, proceed to remove the rounding.
However, it should be noted that the value will not be rounded in the readme, at least so users are informed. 
What do you say?

Reply:
that's fine